### ISSUE TITLE
Make domains for organisations optional

### ISSUE THREAD

**Issue by stonymahony** [keycloak/keycloak#31285]
### Description

When creating an organisation, a domain **must** be specified.

My suggestion would be to make this **optional**.

The domain is currently used to determine which user wants to log in (into which organisation), for example. Without a domain, it could possibly be adapted as follows:

- if there is only one user with the e-mail globally, it is unambiguous
- if there are several, the request must contain the organisation ID or the name, which apparently must be unique; an input field could be added in the ui where the user can enter the name of their organisation

### Motivation

Even if it is rare and becoming increasingly rare, not every organisation (often a "customer" in the SaaS context) has its own domain.

**Comment by nrail-joerg-walter** [comment]
Yes, please. This is way too constrained. I am seeing scenarios where freelancers with gmail addresses are working for a customer organisation. This would be impossible with this structure

**Comment by magoogli** [comment]
In our use case we have a SaaS offering where individuals or companies can sign up for our service. When someone signs up a keycloak 'organization' is created. This is perfect for a multi tenant scenario, where each organization maps to a tenant in our system - but the domain doesn't really have any use for us and is more of an annoyance. At the moment I just auto generate a guid and append .company.com to get around the fact that it is required.

**Comment by Zyles** [comment]
I was surprise to see the domain as a requirement.

We are building a SaaS app as well and this field is not needed.

**Comment by paulwer** [comment]
@pedroigor Is this considdered/wanted for implemantation? We would also realy like to see this beeing optional. We are building an asset management solution with customers mapped as organizations, which are not necessarily having a domain defined.

**Comment by fosple** [comment]
I'm also building a B2B app and was surprised the domain is mandatory to create a new organization. Sadly, this means we cannot use the organizations feature at the current point in time but have to build something custom in our own DB.

The solution of @stonymahony sounds good to me:
>     * if there is only one user with the e-mail globally, it is unambiguous
>     * if there are several, the request must contain the organisation ID or the name, which apparently must be unique; an input 

It's also what big players like Salesforce, Slack & Microsoft do:

<img width="500" alt="Image" src="https://github.com/user-attachments/assets/cf7b952c-a7ad-44bc-9acd-c9e2c8603bdc" />

<img width="500" alt="Image" src="https://github.com/user-attachments/assets/daae8f1c-b4c5-41f8-b0a9-6357f7b03fbd" />

<img width="500" alt="Image" src="https://github.com/user-attachments/assets/30641083-64c4-4d97-b3a5-693289380154" />
